src/linear_algebra/quadratic_form.lean

```diff
@@ -821,3 +822,140 @@ theorem exists_orthogonal_basis [hK : invertible (2 : K)]
 let ⟨v, hv₁, hv₂, _⟩ := exists_orthogonal_basis' hB₁ hB₂ in ⟨v, hv₁, hv₂⟩
 
 end bilin_form
+
+namespace quadratic_form
+
+open_locale big_operators
+
+open finset bilin_form
+
+variables {M₁ : Type*} [add_comm_group M₁] [module R M₁]
+variables {ι : Type*} [fintype ι] {v : ι → M}
+
+instance units.distrib_mul_action : distrib_mul_action (units R) R :=
+{ smul := λ r s, (r : R) * s,
+  one_smul := one_mul,
+  mul_smul := λ _ _, mul_assoc _ _,
+  smul_add := λ _ _, mul_add _ _,
+  smul_zero := λ _, mul_zero _ }
+
+instance : smul_comm_class (units R₁) R₁ R₁ :=
+{ smul_comm := λ _, mul_left_comm _ }
```

Comment:
@eric-wieser I can't seem to find these two instances. Are they somewhere I'm overlooking? If not I will pr them separately.
Also, I have now changed to the generalization version of `weighted_sum_sqaures` though I would now need to say `weighted_sum_squares w : quadratic_form ℂ (ι → ℂ)` instead of just `weighted_sum_squares w`.

Reply:
All these instances are missing, I'm adding them in #7438 but there a lot of cleanup that they're uncovering

Reply:
I recommend you add an explicit argument to `weighed_sum_squares` so that the spelling becomes `weighted_sum_squares ℂ w`, ie "a quadratic form on `C` weighted by W`".